app/components/rails_bootstrap/badge_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class RailsBootstrap::BadgeComponent < ApplicationComponent
+  # rename variant to variation 
+  # variant is a keyword in rails
+  VARIATIONS = ["primary", "secondary", "success", "danger", "warning", "light", "info", "dark"].freeze
+
+  validates :variation, inclusion: { in: VARIATIONS }
+  validates :pill, inclusion: { in: [true, false] }
+
+  def initialize(variation: "primary", pill: false)
+    @variation = variation
+    @pill = pill
+  end
+
+  def styles
+    pill_class = "badge-pill" unless !pill
+    variation_class = "badge-#{variation}"
```

Comment:
Can we extract these two classes as methods. It just helps in separation of concern. Basically something like this: 

```ruby
def styles
  "badge" + pill_class + variation_class
end

private 

def pill_class
  pill ? " badge-pill" : ""
end

def variation_class
  " badge-#{variation}"
end
```